### PR TITLE
Adding nautical rendering for seamarks in INT-1 section K and J 

### PR DESCRIPTION
**Pull request by josail**
INT-1 section K Obstructions and J Seabed, Weed, Seagrass

Adding nautical rendering for seamarks in INT-1 section K Obstructions and J Seabed, Weed, Seagrass, following latest amendment of OpenNauticalChart and OpenSeaMap rendering. 
See especiall: https://wiki.openstreetmap.org/wiki/Seamarks/INT-1_Section_J

including all related .png files for seamark_int1 icons, shaders and additional .svg vector data

OsmAnd nautical.render.xml renderingProperty options for seabedDetail as nameTag2: simple=1st surface type as letters according to INT-1 J1-J13, category=surface tag in seamark_category, all=surface qualification and type included in seabed_surface_all, seagrass and weed in seabed_taxon_all including category, genus and taxon, omit=no display

Tested 2021-12-18 in OffRoad.jar by combination of version 1.1 of rendering_types.xml and version 1.1 of nautical.renderer.xml, e.g.  http://osmand.net/go.html?lat=36.65161&lon=25.37019&z=16, feedback welcome: josail @ gmx.de

Nautical_Rendering_Versions_1_1_INT1_Sektions_K_J_OffRoad_test_result_2021_12_18

### CODE REVIEW

File: rendering_styles/nautical.render.xml [hunk 1]
```diff
@@ -1510,6 +1523,23 @@
 					<apply textWrapWidth="19"/>
 					<apply_if engine_v1="false" textDy="7"/>
 				</switch>
+		
+				<!-- INT-1 J Section: Nature of the Seabed -->
+				<!-- INT-1 J1 to J12 Types of Seabed, OsmAnd nautical.render.xml renderingProperty options for seabedDetail as nameTag2: simple=1st surface type as letters according to INT-1 J1-J13, category=surface tag in seamark_category, all=surface qualification and type included in seabed_surface_all, seagrass and weed in seabed_taxon_all including category, genus and taxon, omit=no display -->
+				<!-- Tested 2021-12-18 together with version 1.1 of rendering_types.xml, e.g.  http://osmand.net/go.html?lat=36.65161&lon=25.37019&z=16, feedback welcome: josail @ gmx.de -->
+				<switch minzoom="14" textWrapWidth="20" textOrder="14" textDy="10" textColor="#0400ab" textHaloColor="#5576cbea" textHaloRadius="2">
+				    <case seabedDetail="simple" tag="seamark:type" value="seabed_area" nameTag="seabed_surface_1"/> 
+					<case seabedDetail="simple" tag="seamark:type" value="weed" nameTag="seabed_letters" icon="seamark_int1j132br_weedkelp_icon"/>
```
Comment (xmd5a2) [inline rendering_styles/nautical.render.xml:1532]: This is a <text> section. "icon" should not be there.

**Comment by xmd5a2** [comment]
SVG in simplified form for icons and shaders should be placed in `icons/svg/` folder into appropriate subdirectory, not into `rendering_styles/style-icons` (this is a folder for automatically generated icons and shaders). Next, all icons (and not shaders) should be added to `icons/tools/sortfiles.sh`. After that `generatepng.sh` and `sortfiles.sh' should be executed. Generated PNGs should be in `rendering_styles/style-icons` directory.

**Comment by josail** [comment]
Dear xmd5a

Thanks for the fast feedback. Will you apply these necessary changes for correctly locating and converting the files? 

The structure in the OsmAnd-Ressources is somewhat different from the OffRoad software which I used for testing. I will be happy to learn the right way of doing it. The location Would be able to manage. At least for the conversion I‘m not sure if I can manage before Christmas. 

If you can do it quickly, I would be happy with your help and to get this done as soon as possible. 

Thanks a lot!

If you like that I do it, then let me know and I try soon as good as I can. 

Best regards and seasonal greeting 

Jo

***@***.***
________________________

Am 20.12.2021 um 09:09 schrieb xmd5a ***@***.***>:

﻿
SVG in simplified form for icons and shaders should be placed in icons/svg/ folder into appropriate subdirectory, not into rendering_styles/style-icons (this is a folder for automatically generated icons and shaders). Next, all icons (and not shaders) should be added to icons/tools/sortfiles.sh. After that generatepng.sh and sortfiles.sh' should be executed. Generated PNGs should be in rendering_styles/style-icons` directory.

**Comment by xmd5a2** [comment]
I've reverted this because there is so much work to adapt this PR
1.
> SVG in simplified form for icons and shaders should be placed in `icons/svg/` folder into appropriate subdirectory, not into `rendering_styles/style-icons` (this is a folder for automatically generated icons and shaders). Next, all icons (and not shaders) should be added to `icons/tools/sortfiles.sh`. After that `generatepng.sh` and `sortfiles.sh' should be executed. Generated PNGs should be in `rendering_styles/style-icons` directory.

2.`spring` SVG icon is empty
3.No support for night mode
4.SVG Icons should be 48x48 size (not 5x10 etc.). Shaders are exception.
5.Please don't give icons such complex names and do as other icons are named.

**Comment by josail** [comment]
Dear xmd5a2. Thanks for the feedback. I have reworked all svg icons and they are ready and as required. I understand that gereatepng.sh and sortfiles.sh automatise a lot of the png and icon creation process. Good to know.  However, I don't manage to get gereatepng.sh running. I did the renaming of the svg's and correspondingly in the nautical.render.xml. 
I will now send this as a second pull request tonight. Perhaps you have the chance to look at it and run ths .sh files. If not, I can try again 2022. Best wishes and seasonal greetings, Jo

**Pull request merged**